### PR TITLE
pipedev 2.3

### PR DESCRIPTION
**Pull request by mlaszloffy**
- A base "MergingDecider" implementation for merging logic
- Workflow run test report now outputs workflow runs and working directories in final report
- Fix SematicWorkflow by packaging a test ontology.conf with the project
- Update project to be compatible with java 8

### CODE REVIEW

File: pipedev-test-utils/src/main/java/ca/on/oicr/pde/dao/executor/PluginExecutor.java [hunk 1]
```diff
@@ -92,7 +93,7 @@ public void workflowRunUpdateStatus(net.sourceforge.seqware.common.model.Workflo
     }
 
     @Override
-    public String workflowRunReport(net.sourceforge.seqware.common.model.WorkflowRun workflowRun) throws IOException {
+    public String workflowRunStatus(net.sourceforge.seqware.common.model.WorkflowRun workflowRun) throws IOException {
```
[the remark this answers is not in the document]
Comment (morgantaschuk) [reply pipedev-test-utils/src/main/java/ca/on/oicr/pde/dao/executor/PluginExecutor.java]: Oh, never mind, there's an implementation in a child class. Is there a better paradigm for things that need implementing than throwing an UnsupportedOperationException? Maybe making the class abstract and leaving the implementation to the child?

File: pipedev-test-utils/src/main/java/ca/on/oicr/pde/dao/executor/ShellExecutor.java [hunk 2]
```diff
@@ -186,17 +185,20 @@ public void workflowRunUpdateStatus(WorkflowRun workflowRun) throws IOException
...
         StringBuilder cmd = new StringBuilder();
         cmd.append("java -cp ").append(getClassPathAsString());
         cmd.append(" io.seqware.cli.Main workflow-run report");
         cmd.append(" --accession ").append(workflowRunSwid.getSwAccession());
 
         File stdOutAndErrFile = new File(loggingDirectory + "/" + "workflowRunReport.out");
 
-        return SeqwareOutputParser.getWorkflowRunStatusFromOutput(
-                Helpers.executeCommand(id, cmd.toString(), workingDirectory, stdOutAndErrFile, environmentVariables));
+        return Helpers.executeCommand(id, cmd.toString(), workingDirectory, stdOutAndErrFile, environmentVariables);
```
Comment (morgantaschuk) [inline pipedev-test-utils/src/main/java/ca/on/oicr/pde/dao/executor/ShellExecutor.java]: Will this return the same as expected even though you've removed the SeqWareOutputParser call? It looks like Helpers is returning CommandResult "output" which may or may not be the workflow run status.

Comment (mlaszloffy) [reply pipedev-test-utils/src/main/java/ca/on/oicr/pde/dao/executor/ShellExecutor.java]: "workflowRunReport" now returns the full report and a new method "workflowRunStatus" returns just the workflow run status.

ca.on.oicr.pde.testing.workflow.OozieWorkflowRunTest has been updated to use the new method "workflowRunStatus".

File: pipedev-test-utils/src/main/java/ca/on/oicr/pde/dao/executor/PluginExecutor.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (mlaszloffy) [reply pipedev-test-utils/src/main/java/ca/on/oicr/pde/dao/executor/PluginExecutor.java]: The majority of testing uses the ShellExecutor implementation of SeqwareExecutor - an implementation that uses the seqware command line via shelling out.

PluginExecutor is an implementation that does not require shelling out, but not all required operations are accessible via the seqware api.  This is an incomplete implementation for now.

File: pipedev-test-utils/pom.xml [hunk 3]
```diff
@@ -173,6 +173,12 @@
             <groupId>ca.on.oicr.gsi</groupId>
             <artifactId>pipedev-provenance-impl</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.assertj</groupId>
+            <artifactId>assertj-core</artifactId>
+            <!-- use 3.8.0 for Java 8 -->
+            <version>2.8.0</version>
```
[the remark this answers is not in the document]
Comment (mlaszloffy) [reply pipedev-test-utils/pom.xml]: Java 8 for pipedev build, but Java 7 for workflow builds + testing.

File: pipedev-test-utils/src/main/java/ca/on/oicr/pde/dao/executor/PluginExecutor.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (morgantaschuk) [reply pipedev-test-utils/src/main/java/ca/on/oicr/pde/dao/executor/PluginExecutor.java]: :+1:

File: pipedev-test-utils/pom.xml (hunk 3 above)
[the remark this answers is not in the document]
Comment (mlaszloffy) [reply pipedev-test-utils/pom.xml]: Or rather, the project is built with compiler source and target set to 1.7.

**Pull request merged**